clippy_lints/src/mod_lib.rs

```diff
@@ -0,0 +1,52 @@
+use clippy_utils::diagnostics::span_lint_and_help;
+use rustc_ast::{ptr::P, Crate, Item, ItemKind, ModKind};
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::sym;
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// ### Why is this bad?
+    ///
+    /// ### Example
+    /// ```rust
+    /// // example code where clippy issues a warning
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// // example code which does not raise clippy warning
+    /// ```
+    #[clippy::version = "1.66.0"]
+    pub MOD_LIB,
+    pedantic,
+    "default lint description"
+}
+declare_lint_pass!(ModLib => [MOD_LIB]);
+
+impl EarlyLintPass for ModLib {
```

Comment:
Hey, thanks for working on this! I know this PR isn’t ready for review, just wanted to say we’re trying to make all new lints to use ‘LateLintPass’, cheers! Feel free to share any of your issues

Reply:
Thanks @kraktus I've updated the code to implement `LateLintPass` instead. Am I going in the right direction here?

Reply:
It does! Looks like your lint is not registered yet, using `cargo new_lint` (don’t remember the exact name of the subcommand, find it with `cargo —help` while in clippy repo, will create the skeleton of the lint for you

Reply:
Actually `cargo dev update_lint` should suffice in the current situation. The lint exists, but it isn't yet registered.